examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -44,6 +48,37 @@ class Task extends TypedEventEmitter<ITaskEvents> implements ITask {
             this.emit("priorityChanged");
         });
     }
+    private readonly resetDiffProperties = (): void => {
+        this.diffPriority = DEFAULT_DIFF_PRIORITY;
+        this.diffType = "none";
+        this.emit('externalPriorityChanged');
+        this.diffName = '';
+        this.diffType = "none";
+        this.emit('externalNameChanged');
+    }
```

Comment:
Something about this still feels funky to me but it shouldn't block merging to the dev branch. I understand now the purpose of the events is to cause re-render, but then I wonder if we could design/name the events a bit differently to achieve that. Mainly I still have hard time with events with these names triggering when the user decided to ignore the external changes.

Reply:
Perhaps a better name for them is actually "diffTypeChanged" and "diffNameChanged" because they do change at each trigger of these, once to populate them and render on screen and if ignoring them, then to clear those values and re-render the clearing on screen. So I think I need to do a passthrough to clean up the names and match them to their event triggers.

Reply:
Couple options:
1. With the naming suggestion above, these could become something like `setSourceName()`
2. Alternatively, if using a setter accessor then the events can be raised from those setters.